src/components/VaultRow/VaultRow.tsx

```diff
@@ -36,29 +52,65 @@ export const VaultRow: FC<VaultTableRowProps> = ({
   showEarningsColumn = false,
   ...props
 }) => {
+  const isCompounderProduct = useIsCompounderProduct(
+    props.productType ?? "compounder"
+  )
+  const compounderProps = props as CompounderVaultProps
+  const concentratorProps = props as ConcentratorVaultProps
   const router = useRouter()
   const { pathname, query } = router
-  const { isLoading } = useVault(props)
+  const { isLoading } = useVault(
```

Comment:
The next thing I would do after the above change is just normalize the data so you don't need ternaries everywhere:

```ts
const asset = isCompounderProduct ? props.asset : props.primaryAsset
const vaultAddress = isCompounderProduct ? props.vaultAddress : props.targetAsset
```

With that you should be able to cleanup a lot of things; starting here:

```ts
const { isLoading } = useVault({ asset, vaultAddress })
```

Reply:
Agree